crates/holochain_state/src/mutations.rs

```diff
@@ -102,16 +90,49 @@ pub fn insert_record_scratch(
     }
 }
 
-/// Insert a [`DhtOp`](holochain_types::dht_op::DhtOp) into the database.
-pub fn insert_op(txn: &mut Transaction, op: &DhtOpHashed) -> StateMutationResult<()> {
-    insert_op_when(txn, op, Timestamp::now())
+/// Insert a [`DhtOp`](holochain_types::dht_op::DhtOp) into the Authored database.
+pub fn insert_op_authored(
+    txn: &mut Ta<DbKindAuthored>,
+    op: &DhtOpHashed,
+) -> StateMutationResult<()> {
+    insert_op_when(txn, op, Timestamp::now(), None)
+}
+
+/// Insert a [`DhtOp`](holochain_types::dht_op::DhtOp) into the DHT database.
+///
+/// If `transfer_data` is None, that means that the Op was locally validated
+/// and is being included in the DHT by self-authority
+pub fn insert_op_dht(
+    txn: &mut Ta<DbKindDht>,
+    op: &DhtOpHashed,
+    transfer_data: Option<(AgentPubKey, TransferMethod, Timestamp)>,
+) -> StateMutationResult<()> {
+    insert_op_when(txn, op, Timestamp::now(), transfer_data)
+}
+
+pub fn insert_op_unchecked(txn: &mut Transaction, op: &DhtOpHashed) -> StateMutationResult<()> {
+    insert_op_when(txn, op, Timestamp::now(), None)
+}
+
+/// Insert a [`DhtOp`](holochain_types::dht_op::DhtOp) into the Cache database.
+pub fn insert_op_cache(
+    txn: &mut Ta<DbKindCache>,
+    op: &DhtOpHashed,
+    transfer_data: Option<(AgentPubKey, TransferMethod, Timestamp)>,
+) -> StateMutationResult<()> {
+    insert_op_when(txn, op, Timestamp::now(), transfer_data)
+}
+
+pub fn todo_no_transfer_data() -> Option<(AgentPubKey, TransferMethod, Timestamp)> {
```

Comment:
What is missing for this to be done?

Reply:
I think it just belongs on a another PR. If I'm understanding right, this PR is supporting the change to track when, how and from who we discovered ops

Reply:
Yes, we share a schema across all 3 op databases, so these fields exist on the cache db but we have no way to populate them -- we don't know who we got ops from yet, when they are cached, that would be a whole lot more extra plumbing.